sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/src/Listeners/QueueScalerProvider.cs

```diff
@@ -87,6 +87,21 @@ public void ResolveProperties(INameResolver resolver)
                 {
                     QueueName = resolver.ResolveWholeString(QueueName);
                 }
+                // Always normalize after token resolution
+                NormalizeQueueName();
+            }
+
+            /// <summary>
+            /// Coerces the resolved queue name to lowercase.
+            /// Azure Queue Storage requires lowercase names; normalization here also
+            /// ensures consistent hashing/identity when used by scaling components.
+            /// </summary>
+            private void NormalizeQueueName()
+            {
+                if (!string.IsNullOrEmpty(QueueName))
+                {
+                    QueueName = QueueName.ToLowerInvariant(); // must be lowercase. coerce here to be nice.
```

Comment:
This might be considered a breaking change. But I'm open to an explanation of how this whole flow is implemented. If the queue name is generated internally and never exposed to external users then I'm fine with it.

Even though the Azure Storage Queue service has a strict requirement that queue names must be lowercase, it's a requirement that is enforced by the Azure Storage Queue service.

Azure SDKs do not automatically lowercase the queue names to avoid silently changing user input, even though the uppercase names are invalid.